src/web5-connect/web5-connect-server.ts

```diff
@@ -83,10 +83,15 @@ export class Web5ConnectServer {
    * Returns the Web5 Connect Request object. The request ID can only be used once.
    */
   public async getWeb5ConnectRequest(requestId: string): Promise<Web5ConnectRequest | undefined> {
-    const request = this.cache.get(`request:${requestId}`);
+    const request = await this.cache.get(`request:${requestId}`);
 
-    // Delete the Request Object from the data store now that it has been retrieved.
-    this.cache.delete(`request:${requestId}`);
+    // Delete the Request Object from cache once it has been retrieved.
+    // IMPORTANT: only delete if object is fetched, otherwise there could be a race condition
+    // where the object is not fetched in this call but become available immediately right,
+    // we would end up deleting it before a successful fetch.
```

Comment:
nit: was not clear to me what "fetched" meant in this context, maybe something like this for a bit more clarity.

```suggestion
    // Delete the Request Object from cache once it has been retrieved.
    // IMPORTANT: only delete if the object exists, otherwise there could be a race condition
    // where the object does not exist in this call but becomes available immediately after,
    // we would end up deleting it before it is successfully retrieved.
```